transcom/mymove
# INT B-21707

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21707)

## Summary

We want the Prime to be able to submit payment requests for basic international HHG service items. This work was kind of a beast because I had to set up pricing for service items that involved some database work and updating the `service_params` & `service_item_param_keys` tables. This PR does the following:
- adds migration to update relevant tables with needed param keys for iHHG service items & other accessorials
- updated the address factory to populate `us_post_region_cities_id` when addresses are created through factory
- added ability for partial weight billing for basic iHHG service items
- added the ability to call the `get_rate_area_id` db func from the `re_rate_area` model file (this is needed to get a lot of pricing data)
- added lookup to get the `per_unit_cents` value from respective international pricing tables - I think this will be good to avoid excessive db calling in the pricing function because we can store it instead of keep looking for it
- added lookup to get the port name that is attached to the `POEFSC` or `PODFSC` service item - I'd like to display this to the TIO when they are reviewing (this work is downstream)
- added pricers for `ISLH`, `IHPK`, `IHUPK`, & `POEFSC`/`PODFSC`
- mocks
- updated prime sim UI to allow for creating payment requests & partial weight billing
- code cleanup & tests

### How to test

1. Go ahead and create a move as a customer/SC with an `iHHG` shipment
2. Approve as TOO to make it available to the prime
3. Update the estimated weight and all required data
4. At this point you can create a payment request for `ISLH`, `IHPK`, and `IHUPK` service items - we need to update the port to get the payment request working for `POEFSC`/`PODFSC`
5. You're gonna need two things: the `eTag` for the fuel surcharge service item (Prime v3 `getMoveTaskOrder`) & the service item `id` - [swagger UI makes it easy](http://primelocal:3000/swagger-ui/prime_v3.html#/moveTaskOrder/getMoveTaskOrder)
6. Now you will need to update the port of the service item by using [prime v1 updateServiceItem](http://officelocal:3000/swagger-ui/prime.html#/mtoServiceItem/updateMTOServiceItem) - make sure you got that `eTag`
7. Payload can look like this (using your respective values):
```
{   
   "id": "4cc44e88-231f-4f49-8180-4d7012be17b3",   
   "modelType": "UpdateMTOServiceItemInternationalPortFSC",   
   "portCode": "SEA",   
   "reServiceCode": "POEFSC" 
}
```
8. You could also just update the db manually, but it's up to you
9. Now go ahead and create a payment request for all basic service items (or just the fuel surcharge if you did the others already)
10. Everything should be successful and you should be able to review as TIO - breakdown/details are coming in a downstream BL
11. Verify you can also see the updated documentation: [use swagger](http://officelocal:3000/swagger-ui/prime.html#/paymentRequest/createPaymentRequest)

## Screenshots
![Screenshot 2025-01-02 at 4 41 07 PM](https://github.com/user-attachments/assets/1185a188-c0e6-4c3f-a32b-a0b74ebf7977)
![Screenshot 2025-01-02 at 4 41 20 PM](https://github.com/user-attachments/assets/8f7d96e0-c204-47df-8c11-abe9f2088155)
![Screenshot 2025-01-02 at 4 41 24 PM](https://github.com/user-attachments/assets/ab5f7930-d644-4172-bb21-dd1feb9ef5f4)
